cpp/density/RDF.cc

```diff
@@ -28,262 +27,192 @@ RDF::RDF(float rmax, float dr, float rmin)
-        m_vol_array3D.get()[i] = 4.0f / 3.0f * M_PI * (nextr*nextr*nextr - r*r*r);
-        }
+        float nextr = float(i + 1) * m_dr + m_rmin;
+        m_r_array.get()[i] = 2.0f / 3.0f * (nextr * nextr * nextr - r * r * r) / (nextr * nextr - r * r);
+        m_vol_array2D.get()[i] = M_PI * (nextr * nextr - r * r);
+        m_vol_array3D.get()[i] = 4.0f / 3.0f * M_PI * (nextr * nextr * nextr - r * r * r);
+    }
 
-    }  // end RDF::RDF
+} // end RDF::RDF
 
 RDF::~RDF()
+{
+    for (tbb::enumerable_thread_specific<unsigned int*>::iterator i = m_local_bin_counts.begin();
+         i != m_local_bin_counts.end(); ++i)
     {
-    for (tbb::enumerable_thread_specific<unsigned int *>::iterator i = m_local_bin_counts.begin(); i != m_local_bin_counts.end(); ++i)
-        {
-        delete[] (*i);
-        }
+        delete[](*i);
     }
+}
 
 //! \internal
 //! CumulativeCount class to perform a parallel reduce to get the cumulative count for each histogram bin
 class CumulativeCount
+{
+private:
+    float m_sum;
+    float* m_N_r_array;
+    float* m_avg_counts;
+public:
+    CumulativeCount( float *N_r_array,
+          float *avg_counts )
+        : m_sum(0), m_N_r_array(N_r_array), m_avg_counts(avg_counts)
     {
-    private:
-        float m_sum;
-        float* m_N_r_array;
-        float* m_avg_counts;
-    public:
-        CumulativeCount( float *N_r_array,
-              float *avg_counts )
-            : m_sum(0), m_N_r_array(N_r_array), m_avg_counts(avg_counts)
-        {
-        }
-        float get_sum() const
-            {
-            return m_sum;
-            }
-        template<typename Tag>
-        void operator()( const blocked_range<size_t>& r, Tag )
-            {
-            float temp = m_sum;
-            for( size_t i=r.begin(); i<r.end(); i++ )
-                {
-                temp = temp + m_avg_counts[i];
-                if( Tag::is_final_scan() )
-                    m_N_r_array[i] = temp;
-                }
-            m_sum = temp;
-            }
-        CumulativeCount( CumulativeCount& b, split )
-            : m_sum(0), m_N_r_array(b.m_N_r_array), m_avg_counts(b.m_avg_counts)
+    }
+    float get_sum() const
+    {
+        return m_sum;
+    }
+    template<typename Tag>
+    void operator()( const blocked_range<size_t>& r, Tag )
+    {
+        float temp = m_sum;
+        for( size_t i=r.begin(); i<r.end(); i++ )
         {
+            temp = temp + m_avg_counts[i];
+            if( Tag::is_final_scan() )
+                m_N_r_array[i] = temp;
         }
-        void reverse_join( CumulativeCount& a )
-            {
-            m_sum = a.m_sum + m_sum;
-            }
-        void assign( CumulativeCount& b )
-            {
-            m_sum = b.m_sum;
-            }
-    };
-
+        m_sum = temp;
+    }
+    CumulativeCount( CumulativeCount& b, split )
+        : m_sum(0), m_N_r_array(b.m_N_r_array), m_avg_counts(b.m_avg_counts)
+    {
+    }
```

Comment:
Does our clang style suggest always splitting this line? Just wondering.

Reply:
Seems like it does not split it when I tried right now.